kubetest/eks/eks.go

```diff
@@ -0,0 +1,272 @@
+	_, err = exec.LookPath("aws-iam-authenticator")
+	if err != nil {
+		return nil, fmt.Errorf("cannot find 'aws-iam-authenticator' executable (%v)", err)
+	}
+
+	return tr, nil
+}
+
+// Up creates a new EKS cluster.
+func (tr *tester) Up() (err error) {
+	createCmd := exec.Command(
+		tr.awsTesterPath,
+		"eks",
+		"--path="+tr.cfg.ConfigPath,
+		"create",
+		"cluster",
+	)
+	errc := make(chan error)
+	go func() {
+		ctrl := process.NewControl(
+			tr.cfg.KubetestControlTimeout,
+			time.NewTimer(tr.cfg.KubetestControlTimeout),
+			time.NewTimer(tr.cfg.KubetestControlTimeout),
+			tr.cfg.KubetestVerbose,
+		)
+		_, oerr := ctrl.Output(createCmd)
+		errc <- oerr
+	}()
+	select {
+	case <-tr.stopc:
+		fmt.Fprintln(os.Stderr, "received stop signal, interrupting 'create cluster' command...")
+		ierr := createCmd.Process.Signal(syscall.SIGINT)
+		err = fmt.Errorf("'create cluster' command interrupted (interrupt error %v)", ierr)
+	case err = <-errc:
+	}
+	return err
+}
+
+// Down tears down the existing EKS cluster.
+func (tr *tester) Down() (err error) {
+	ctrl := process.NewControl(
+		tr.cfg.KubetestControlTimeout,
+		time.NewTimer(tr.cfg.KubetestControlTimeout),
+		time.NewTimer(tr.cfg.KubetestControlTimeout),
+		tr.cfg.KubetestVerbose,
+	)
+	_, err = ctrl.Output(exec.Command(
+		tr.awsTesterPath,
+		"eks",
+		"--path="+tr.cfg.ConfigPath,
+		"delete",
+		"cluster",
+	))
+	return err
+}
+
+// IsUp returns an error if the cluster is not up and running.
+func (tr *tester) IsUp() (err error) {
+	ctrl := process.NewControl(
+		tr.cfg.KubetestControlTimeout,
+		time.NewTimer(tr.cfg.KubetestControlTimeout),
+		time.NewTimer(tr.cfg.KubetestControlTimeout),
+		tr.cfg.KubetestVerbose,
+	)
+	_, err = ctrl.Output(exec.Command(
+		tr.awsTesterPath,
+		"eks",
+		"--path="+tr.cfg.ConfigPath,
+		"check",
+		"cluster",
+	))
+	if err != nil {
+		return err
+	}
+
+	tr.cfg, err = eksconfig.Load(tr.cfg.ConfigPath)
+	if err != nil {
+		return err
+	}
+
+	if tr.cfg.ClusterState.Status != "ACTIVE" {
+		return fmt.Errorf("cluster %q status is %q",
+			tr.cfg.ClusterName,
+			tr.cfg.ClusterState.Status,
+		)
+	}
+	return nil
+}
+
+// TestSetup checks if EKS testing cluster has been set up or not.
+func (tr *tester) TestSetup() error {
+	return tr.IsUp()
+}
+
+// GetClusterCreated returns EKS cluster creation time and error (if any).
+func (tr *tester) GetClusterCreated(v string) (time.Time, error) {
+	err := tr.IsUp()
+	if err != nil {
+		return time.Time{}, err
+	}
+	tr.cfg, err = eksconfig.Load(tr.cfg.ConfigPath)
+	if err != nil {
+		return time.Time{}, err
+	}
+	return tr.cfg.ClusterState.Created, nil
+}
+
+// DumpClusterLogs uploads local cluster logs to S3.
+func (tr *tester) DumpClusterLogs(localPath, s3Path string) error {
+	if !tr.cfg.KubetestEnableDumpClusterLogs {
+		// let default kubetest log dumper handle all artifact uploads.
+		// See https://github.com/kubernetes/test-infra/pull/9811/files#r225776067.
```

Comment:
our util will handle your pod logs, but you'll still need to generate things like kubelet logs and master logs somehow :-)

@justinsb already wrote https://github.com/kubernetes/test-infra/blob/master/kubetest/dump.go for kops-aws, maybe you can utilize that?